src/Compilers/CSharp/Portable/Declarations/Declaration.cs

```diff
@@ -61,6 +61,11 @@ public ImmutableArray<Declaration> Children
         }
 
         public abstract DeclarationKind Kind { get; }
+
         protected abstract ImmutableArray<Declaration> GetDeclarationChildren();
+
+        public bool IsNamespace => Kind == DeclarationKind.Namespace;
+
+        public bool IsType => !IsNamespace;
```

Comment:
i would like INamespaceOrTypeDeclaration to have either a `Children` or `ChildDeclarations` property that returns an `ImmutableArray<INamespaceOrTypeDeclaratoin>`.

Reply:
I was thinking something different, proposing that `INamespaceOrTypeDeclaration` would not have `Children`, but instead `INamespaceDeclaration` and `ITypeDeclaration` would. 
It doesn't seem useful to get a set of `INamespaceOrTypeDeclaration` from an `ITypeDeclaration`, even if it was casted to the abstract base.

---
In reply to: [266261027](https://github.com/dotnet/roslyn/pull/34067#discussion_r266261027) [](ancestors = 266261027)